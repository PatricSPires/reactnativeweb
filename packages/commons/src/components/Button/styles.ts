import styled, { css } from 'styled-components/native'
import { ThemeProps } from '../../styles/customThemes/types'

import { ButtonTextProps, ButtonWrapperProps } from './types'
import { getButtonAppearance, getTextAppearance } from './appearance'

export const ButtonWrapper = styled.TouchableOpacity<
  ButtonWrapperProps & ThemeProps
>`
  ${({ alignIcon, alignItems, theme }) => css`
    display: flex;
    align-items: center;
    height: 48px;
    max-width: 300px;
    min-width: 48px;
    width: 178px;
    border: none;
    padding: ${theme.spacings[2]};
    flex-direction: ${alignIcon === 'left' ? 'row-reverse' : 'row'};
    justify-content: ${alignItems === 'center' ? 'center' : 'space-between'};
    ${getButtonAppearance}
  `}
`

export const ButtonText = styled.Text<ButtonTextProps>`
  ${getTextAppearance}
`
