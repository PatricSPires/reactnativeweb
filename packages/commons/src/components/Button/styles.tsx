import React from 'react'
import styled, { css } from 'styled-components'
import { Platform, TouchableOpacity, Text } from 'react-native'
import { ButtonProps } from './types'
import { buttonAppearances, textAppearances } from './appearance'

type ButtonComponentProps = Pick<
  ButtonProps,
  | 'appearance'
  | 'marginBottom'
  | 'width'
  | 'alignIcon'
  | 'alignItems'
  | 'className'
>
type TextComponentProps = Pick<ButtonProps, 'alignText' | 'appearance'>

type ButtonComponent1Props = {
  className?: string
  children: JSX.Element
  onPress: () => void
}

const Button = styled(TouchableOpacity)`
  ${({ appearance }: ButtonComponentProps) => css`
    border: none;
    display: flex;
    flex-direction: ${({ alignIcon }: ButtonComponentProps) =>
      alignIcon === 'left' ? 'row-reverse' : 'row'};
    align-items: center;
    justify-content: ${({ alignItems }: ButtonComponentProps) =>
      alignItems === 'center' ? 'center' : 'space-between'};
    height: 48px;
    max-width: 300px;
    min-width: 48px;
    width: 178px;
    padding: 16px;

    ${Platform.OS === 'web' && `cursor: pointer;`}
    ${!!appearance && buttonAppearances[appearance]()};
  `}
`
const ButtonComponent = ({
  onPress,
  className,
  children,
  ...props
}: ButtonComponent1Props) => {
  // eslint-disable jsx-props-no-spreading
  return (
    <Button
      className={className}
      {...props}
      as={Platform.OS === 'web' ? 'button' : TouchableOpacity}
      onClick={onPress}
      onPress={onPress}
    >
      {children}
    </Button>
  )
}

ButtonComponent.defaultProps = {
  className: null,
}

export const TextComponent = styled(Text)<TextComponentProps>`
  ${({ appearance }: TextComponentProps) => css`
    ${!!appearance && textAppearances[appearance]()};
  `}
`
