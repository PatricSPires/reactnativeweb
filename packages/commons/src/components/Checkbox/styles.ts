import { Text, TouchableWithoutFeedback, View } from 'react-native'
import styled from 'styled-components/native'
import theme from '../../styles/theme'
import { CheckboxProps } from './types'

export const CheckBoxWrapper = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: ${theme.colors.high_contrast_border};
  border-radius: 2px;
  display: flex;
  align-items: center;
  justify-content: center;

  img {
    width: 15px;
    height: 15px;
  }

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === false) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }

    if (checked === false && disabled === true) {
      return `
        border-color: ${theme.colors.primary_subtle_border};
        background-color: transparent;
      `
    }
    return ``
  }}
`

export const CheckBoxCheck = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === true) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }
    return ``
  }}
`

export const CheckBoxText = styled(Text)`
  margin-left: 8px;
  color: ${(props: CheckboxProps) =>
    props.disabled
      ? theme.colors.primary_subtle_border
      : theme.colors.high_contrast_border};
`

export const Container = styled(TouchableWithoutFeedback)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`
