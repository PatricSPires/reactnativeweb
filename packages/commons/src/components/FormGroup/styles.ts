import styled from 'styled-components'
import { View, Text } from 'react-native'
import theme from '../../styles/theme'
import { FormGroupWrapperProps } from './types'

export const FormGroupWrapper = styled(View)<FormGroupWrapperProps>`
  display: flex;
  width: ${({ width }) => width || '100%'};
  margin-bottom: ${({ marginBottom }) => theme.spacings[marginBottom || 3]};

  Tag {
    margin: 8px 0 4px 0;
  }
`

export const Label = styled(Text)`
  color: ${(props) => props.theme.colors.font_color};
  font-size: ${theme.fontSize.label_01};
  margin-bottom: ${theme.spacings[1]};
  display: flex;
  align-items: center;
`

export const HelperText = styled(Text)`
  font-size: ${theme.fontSize.label_01};
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: ${theme.fontSize.label_01};
  color: ${theme.supportColors.error};
  margin-top: ${theme.spacings[1]};
`
