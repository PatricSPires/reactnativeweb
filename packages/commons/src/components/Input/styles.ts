import styled, { css } from 'styled-components'
import {
  Text,
  TextInput as InputRN,
  View,
  TouchableWithoutFeedback,
} from 'react-native'
import { TextInputProps, IconProps, InputWrapperProps } from './types'
import theme from '../../styles/theme'

function getFocusStyles(isInvalid: boolean) {
  if (isInvalid) {
    return `
      outline: 2px solid ${theme.supportColors.error};
      border-bottom-color: transparent;
    `
  }
  return `
    outline: 2px solid ${theme.colors.primary_background};
    border-bottom-color: transparent;
  `
}

export const Wrapper = styled(View)`
  width: 100%;
`

export const InputWrapper = styled(View)<InputWrapperProps>`
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${(props) =>
    props.theme.colors.primary_container_background};
  border-bottom: 1px solid ${theme.colors.medium_contrast_border};

  ${({ isInvalid }) => css`
    outline: ${!!isInvalid && `2px solid ${theme.supportColors.error}`};
    border: ${!!isInvalid && 'none'};
  `}

  &:focus-within {
    ${({ isInvalid }) => getFocusStyles(isInvalid)}
  }
`

export const TextInput = styled(InputRN)<TextInputProps>`
  width: 100%;
  height: 40px;
  color: ${(props) => props.theme.colors.font_color};
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;

  order: ${({ iconPosition }) => (iconPosition === 'right' ? 0 : 1)};
  padding: ${({ iconPosition }) =>
    iconPosition === 'left' ? '0 10px 0 0' : '0 0 0 10px'};

  &:disabled {
    outline: none;
    border: none;
    color: ${(props) => props.theme.colors.font_color};
  }

  ::placeholder {
    color: ${(props) => props.theme.colors.font_color};
  }

  &:focus-within {
    color: ${(props) => props.theme.colors.font_color};
  }
`

export const HelperText = styled(Text)`
  font-size: 12px;
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: 12px;
  color: ${theme.supportColors.error};
`

export const Icon = styled(View)<IconProps>`
  order: ${({ iconPosition }) => (iconPosition === 'right' ? 1 : 0)};
  color: ${theme.colors.primary_background};
  margin: 0 10px;

  svg {
    width: 16px;
    ${({ iconColor }: IconProps) => css`
      color: ${theme.supportColors[iconColor]};
    `}
  }
`

export const PasswordIconWrapper = styled(TouchableWithoutFeedback)`
  width: 16px;
  height: 16px;
  color: ${(props) => props.theme.colors.font_color};
  margin: 0 10px;
`
