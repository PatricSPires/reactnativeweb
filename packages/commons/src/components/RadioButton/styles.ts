import styled from 'styled-components'
import { TouchableWithoutFeedback, View, Text } from 'react-native'
import theme from '../../styles/theme'
import { RadioButtonProps } from './types'

export const RadioButtonWrapper = styled(View)`
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: ${theme.colors.high_contrast_border};
  border-radius: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  ${({ checked, disabled }: RadioButtonProps) => {
    if (checked === false && disabled) {
      return `
        border-color: ${theme.colors.primary_subtle_border};
      `
    }
    return ``
  }}
`

export const RadioButtonCheck = styled(View)`
  width: 8px;
  height: 8px;
  border-radius: 100%;

  ${({ checked, disabled }: RadioButtonProps) => {
    if (checked && checked === true && !disabled) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }

    if (checked && checked === true && disabled && disabled === true) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }
    return ``
  }}
`

export const RadioButtonText = styled(Text)`
  margin-left: 8px;
  color: ${(props: RadioButtonProps) =>
    props.disabled && props.disabled === true
      ? theme.colors.primary_subtle_border
      : theme.colors.high_contrast_border};
`

export const Container = styled(TouchableWithoutFeedback)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: ${(props: RadioButtonProps) =>
    props.disabled && props.disabled === true ? 'not-allowed' : 'pointer'};

  &:focus ${RadioButtonWrapper} {
    border-color: ${theme.supportColors.information};
    box-shadow: 0px 0px 0px 1px ${theme.supportColors.information};
    ${({ checked, disabled }: RadioButtonProps) => {
      if (checked === true && disabled === false) {
        return `
          box-shadow: 0px 0px 0px 1px ${theme.supportColors.information};
        `
      }
      return ``
    }}
  }
`
