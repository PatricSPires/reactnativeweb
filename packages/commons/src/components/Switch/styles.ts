import styled, { css } from 'styled-components'
import { TouchableOpacity, TouchableWithoutFeedback } from 'react-native'
import { SwitchProps } from './types'
import theme from '../../styles/theme'

export const Container = styled(TouchableOpacity)`
  position: relative;
  display: inline-block;
  width: ${(props: SwitchProps) => (props.size === 'small' ? 32 : 52)}px;
  height: ${(props: SwitchProps) => (props.size === 'small' ? 16 : 24)}px;
  border-radius: ${(props: SwitchProps) =>
    props.size === 'small' ? 100 : 12}px;
  background-color: ${theme.colors.medium_contrast_border};
  outline: 2px solid;
  outline-color: transparent;

  ${(props: SwitchProps) =>
    props.checked &&
    css`
      background-color: ${theme.supportColors.success};

      &:focus {
        outline-color: ${theme.supportColors.information};
      }
    `}

  ${(props: SwitchProps) =>
    props.disabled &&
    css`
      background-color: ${theme.colors.primary_subtle_border};
      outline-color: transparent;
    `}
`

export const Wrapper = styled(TouchableWithoutFeedback)`
  position: absolute;
  width: ${(props: SwitchProps) => (props.size === 'small' ? 10 : 18)}px;
  height: ${(props: SwitchProps) => (props.size === 'small' ? 10 : 18)}px;
  border-radius: 50%;
  background-color: #fff;
  top: 3px;
  left: 3px;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;

  ${({ checked, size }: SwitchProps) =>
    checked &&
    `
    -webkit-transition: left 0.25s ease;
    transition: left 0.25s ease;
    left: ${size === 'small' ? 19 : 31}px;
  `}

  ${({ disabled }) =>
    disabled &&
    `
    background-color: ${theme.colors.medium_contrast_border};
  `}
`
