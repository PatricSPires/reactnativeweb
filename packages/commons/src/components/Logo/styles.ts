import styled from 'styled-components/native'
import { ThemeProps } from '../../styles/types'

const LogoWrapper = styled.View`
  margin-bottom: ${({ theme }: ThemeProps) => theme.spacings[6]};
`
export default LogoWrapper
