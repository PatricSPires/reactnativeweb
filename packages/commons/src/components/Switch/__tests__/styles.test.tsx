import React from 'react'
import { Text } from 'react-native'
import { render } from '@testing-library/react-native'
import { Container, Wrapper } from '../styles'
import 'jest-styled-components'

describe('Container > Style', () => {
  it('should render the Container with size small', () => {
    const { toJSON } = render(<Container size="small" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Container with checked true', () => {
    const { toJSON } = render(<Container checked />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Container with disabled true', () => {
    const { toJSON } = render(<Container disabled />)
    expect(toJSON()).toMatchSnapshot()
  })
})

describe('Wrapper > Style', () => {
  it('should render the Wrapper with size small', () => {
    const { toJSON } = render(
      <Wrapper size="small">
        <Text />
      </Wrapper>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Wrapper with size small and checked true', () => {
    const { toJSON } = render(
      <Wrapper size="small" checked>
        <Text />
      </Wrapper>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Wrapper with size large and checked', () => {
    const { toJSON } = render(
      <Wrapper checked size="large">
        <Text />
      </Wrapper>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Wrapper with disabled true', () => {
    const { toJSON } = render(
      <Wrapper disabled>
        <Text />
      </Wrapper>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
