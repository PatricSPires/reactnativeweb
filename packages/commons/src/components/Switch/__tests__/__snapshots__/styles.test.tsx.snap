// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container > Style should render the Container with checked true 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 52px;
  height: 24px;
  border-radius: 12px;
  background-color: #8d8d8d;
  outline: 2px solid;
  outline-color: transparent;
  background-color: #24A148;
}

.c0:focus {
  outline-color: #0043CE;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  tabIndex="0"
/>
`;

exports[`Container > Style should render the Container with disabled true 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 52px;
  height: 24px;
  border-radius: 12px;
  background-color: #8d8d8d;
  outline: 2px solid;
  outline-color: transparent;
  background-color: #e0e0e0;
  outline-color: transparent;
}

<div
  aria-disabled={true}
  className="c0 sc-bdnylx c0 css-view-1dbjc4n r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
/>
`;

exports[`Container > Style should render the Container with size small 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 32px;
  height: 16px;
  border-radius: 100px;
  background-color: #8d8d8d;
  outline: 2px solid;
  outline-color: transparent;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  tabIndex="0"
/>
`;

exports[`Wrapper > Style should render the Wrapper with disabled true 1`] = `
.c0 {
  position: absolute;
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: #fff;
  top: 3px;
  left: 3px;
  -webkit-transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;
  background-color: #8d8d8d;
}

<div
  aria-disabled={true}
  className="c0 sc-gtssRu c0 css-text-901oao"
  dir="auto"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
/>
`;

exports[`Wrapper > Style should render the Wrapper with size large and checked 1`] = `
.c0 {
  position: absolute;
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: #fff;
  top: 3px;
  left: 3px;
  -webkit-transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;
  left: 31px;
}

<div
  checked={true}
  className="c0 sc-gtssRu c0 css-text-901oao r-cursor-1loqt21 r-touchAction-1otgn73"
  data-focusable={true}
  dir="auto"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="large"
  tabIndex="0"
/>
`;

exports[`Wrapper > Style should render the Wrapper with size small 1`] = `
.c0 {
  position: absolute;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: #fff;
  top: 3px;
  left: 3px;
  -webkit-transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;
}

<div
  className="c0 sc-gtssRu c0 css-text-901oao r-cursor-1loqt21 r-touchAction-1otgn73"
  data-focusable={true}
  dir="auto"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="small"
  tabIndex="0"
/>
`;

exports[`Wrapper > Style should render the Wrapper with size small and checked true 1`] = `
.c0 {
  position: absolute;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: #fff;
  top: 3px;
  left: 3px;
  -webkit-transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  -webkit-transition: left 0.25s ease;
  transition: left 0.25s ease;
  left: 19px;
}

<div
  checked={true}
  className="c0 sc-gtssRu c0 css-text-901oao r-cursor-1loqt21 r-touchAction-1otgn73"
  data-focusable={true}
  dir="auto"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="small"
  tabIndex="0"
/>
`;
